### ISSUE TITLE
Linux: Error when trying to select a system

### ISSUE THREAD

**Issue by brightlego** [EDDiscovery/EDDiscovery#3234]
### Description
Trying to select a system in the 'Show System' Menu on Linux (through mono)

### Additional Information
Not much more to add

### Exception Details:
>EDDiscovery v12.1.4.0 There was an unhandled UI exception.
>```
>==== BEGIN ====
>System.NullReferenceException: Object reference not set to an instance of an object
>  at ExtendedControls.ExtTextBoxAutoComplete.cbdropdown_SelectedIndexChanged (System.Object sender, System.EventArgs e) [0x00001] in <cc43136c618c4167872f8466d486817f>:0 
>  at ExtendedControls.ExtListBoxForm.listcontrol_SelectedIndexChanged (System.Object sender, System.EventArgs e) [0x0001f] in <cc43136c618c4167872f8466d486817f>:0 
>  at ExtendedControls.ExtListBox.lbsys_SelectedIndexChanged (System.Object sender, System.EventArgs e) [0x0002c] in <cc43136c618c4167872f8466d486817f>:0 
>  at System.Windows.Forms.ListBox.OnSelectedIndexChanged (System.EventArgs e) [0x00020] in <6d635ac3dc1c4424ad385ded79f1e868>:0 
>  at System.Windows.Forms.ListBox.OnItemClick (System.Int32 index) [0x00000] in <6d635ac3dc1c4424ad385ded79f1e868>:0 
>  at System.Windows.Forms.ListBox.OnMouseUpLB (System.Object sender, System.Windows.Forms.MouseEventArgs e) [0x00063] in <6d635ac3dc1c4424ad385ded79f1e868>:0 
>  at System.Windows.Forms.Control.OnMouseUp (System.Windows.Forms.MouseEventArgs e) [0x00019] in <6d635ac3dc1c4424ad385ded79f1e868>:0 
>  at System.Windows.Forms.Control.WmLButtonUp (System.Windows.Forms.Message& m) [0x00078] in <6d635ac3dc1c4424ad385ded79f1e868>:0 
>  at System.Windows.Forms.Control.WndProc (System.Windows.Forms.Message& m) [0x001b4] in <6d635ac3dc1c4424ad385ded79f1e868>:0 
>  at System.Windows.Forms.ListBox.WndProc (System.Windows.Forms.Message& m) [0x0003e] in <6d635ac3dc1c4424ad385ded79f1e868>:0 
>  at System.Windows.Forms.Control+ControlWindowTarget.OnMessage (System.Windows.Forms.Message& m) [0x00000] in <6d635ac3dc1c4424ad385ded79f1e868>:0 
>  at System.Windows.Forms.Control+ControlNativeWindow.WndProc (System.Windows.Forms.Message& m) [0x0000b] in <6d635ac3dc1c4424ad385ded79f1e868>:0 
>  at System.Windows.Forms.NativeWindow.WndProc (System.IntPtr hWnd, System.Windows.Forms.Msg msg, System.IntPtr wParam, System.IntPtr lParam) [0x00085] in <6d635ac3dc1c4424ad385ded79f1e868>:0 
>===== END =====
>```

**Comment by klightspeed** [comment]
`SelectedIndexChanged` probably needs to check that `cbdropdown` is not null, since it sounds like `SelectedIndexChanged` events are queued in Mono's `System.Windows.Forms` implementation.

**Comment by brightlego** [comment]
I was clicking on an item in the dropdown

**Comment by klightspeed** [comment]
Replicated this by clicking in dropdown, which hadn't been dismissed when I typed a new system name.

* The dropdown steals focus (I suspect that it gets focused after the program returns to the message loop, which is after the `Focus()` call in [`ExtTextBoxAutoComplete.AutoCompleteFinished@TextBoxAutoComplete.cs:195`](https://github.com/EDDiscovery/ExtendedControls/blob/8cb6bd34ddaf657138c20668185948187a06a7eb/ExtendedControls/Controls/TextBoxAutoComplete.cs#L195)
* Clicking back in the text box calls `Close()` in [`ExtTextBoxAutoComplete.CancelAutoComplete@TextBoxAutoComplete.cs:90`](https://github.com/EDDiscovery/ExtendedControls/blob/8cb6bd34ddaf657138c20668185948187a06a7eb/ExtendedControls/Controls/TextBoxAutoComplete.cs#L90) (without unhooking the `SelectedIndexChanged` event)
* Clicking on the zombie dropdown invokes the `SelectedIndexChanged` handler, now with `cbdropdown` null

Closing: it seems that Mono requires that `Dispose` is called in order for a form to close properly.
Focus: `Form.WmActivate` focuses the active control on the form it's received on, thus stealing focus.  The `Activated` event is called after this, where focus can be stolen back.
